src/Generator/PropertyGenerator.php

```diff
@@ -159,6 +171,47 @@ public function isConst()
         return (bool) ($this->flags & self::FLAG_CONSTANT);
     }
 
+    /**
+     * @param  bool $readonly
+     * @return PropertyGenerator
+     */
+    public function setReadonly($readonly)
+    {
+        if ($readonly) {
+            $this->setFlags(self::FLAG_READONLY);
+        } else {
+            $this->removeFlag(self::FLAG_READONLY);
+        }
+
+        return $this;
+    }
+
+    /**
+     * @return bool
+     */
+    public function isReadonly()
+    {
+        return (bool) ($this->flags & self::FLAG_READONLY);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function setFlags($flags)
```

Comment:
Flags are quite a low-level representation of this: let's make this method `private`, if possible

Reply:
This is just an override of the parent in order to throw exceptions for impossible flag combinations